interface/src/app/sample/sampleDetail.component.js

```diff
@@ -0,0 +1,71 @@
+angular.module('adage.sampleDetail', [
+  'adage.sample.service',
+  'adage.experiment.service',
+  'statusBar'
+])
+
+.component('sampleDetail', {
+  templateUrl: 'sample/sampleDetail.component.tpl.html',
+  bindings: {
+    id: '<',
+    onLoad: '&'
+  },
+  controller: [
+    'Sample', 'Experiment', '$log',
+    function(Sample, Experiment, $log) {
+      var ctrl = this;
+      ctrl.makeHref = Experiment.makeHref;
+
+      ctrl.$onChanges = function(changesObj) {
+        ctrl.show(changesObj.id.currentValue);
+      };
+      ctrl.sample = {
+        status: '',
+        results: {},
+        relatedExperiments: []
+      };
+
+      var queryError = function(responseObject) {
+        $log.warn(
+          'adage.sampleDetail: query errored with: ' + responseObject
+        );
+        ctrl.sample.status = 'Query failed.';
+      };
+
+      ctrl.show = function(id) {
+        if (!id) {
+          $log.warn('adage.sampleDetail: show() called with id', id);
```

Comment:
That space would be needed if the arguments to `$log.warn()` were concatenated into one string, but the comma syntax (passing the variable as a second argument) actually allows us to view an object in the console, which I've found to be better because we don't need to "flatten" the object with something like JSON.stringify first.

Reply:
Ah, good point about the falsy, aspect though. I think it would still work as is. There are several ways to make something "falsy" and if we hit this error condition it will be helpful to know exactly what type of falsy thing we're looking at... what do you think?

Reply:
@mhuyck: You are definitely right about the `log.warn()` call. I forgot that it would add delimiter automatically. You can leave the code as is. Thanks.